subprojects/tooling-api/src/main/java/org/gradle/tooling/TestPatternSpec.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling;
+
+import org.gradle.api.Incubating;
+
+import java.util.Collection;
+
+/**
+ * Provides fine-grained test selection within a test task.
+ *
+ * @since 7.5
+ */
+@Incubating
+public interface TestPatternSpec {
+
+
+
+    TestPatternSpec includePackage(String pkg);
+    TestPatternSpec includePackages(Collection<String> packages);
```

Comment:
We should be clear if includePackage means include _only_ this package or this package and any subpackages (which I think is the way it's implemented?)

Reply:
No, it's recursive at the moment. It's consistent with the command line where it's only possible to run tests from a package with the tests from the sub-packages. It would be nice to support the non-recursive case, but maybe in a separate PR.

Reply:
I've checked the current behavior of IntelliJ. It turns out that if you run all tests from a package, the sub-packages will be also included in the test execution.